index/src/index.rs

```diff
@@ -64,14 +66,21 @@ impl Index {
         Ok(())
     }
 
-    pub fn index(self) -> TantivyIndex {
-        self.index
+    pub fn index(&self) -> &TantivyIndex {
+        &self.index
     }
 
     pub fn schema(self) -> SearchSchema {
         self.schema
     }
 
+    pub fn get_page_body(&self, page: u32, path: &str) -> Result<String> {
+        let doc = PdfDocument::load(format!("{}/{}", self.path, path))
+            .map_err(|_e| PdfNotFoundError(format!("{}/{}", self.path, path)))?;
+        let text = doc.extract_text(&[page]).unwrap();
```

Comment:
Here, we should also return an error instead and leave the handling to the caller's context

Reply:
You're referring to line 80 (`let text = doc.extract_text(&[page]).unwrap();`) right?

Reply:
Yes!